### ISSUE TITLE
[Workloads] Unable to find exposed port for Emojivoto web service

### ISSUE THREAD

**Issue by leecalcote** [layer5io/meshery-linkerd#109]
**Description**
Meshery adapter for Linkerd will attempt to retrieve the service port that has been exposed upon deployment of a given workload (sample app).

**Expected Behavior**
When provisioning Emojivoto, users should be told what port its "web-svc" is available on.

**Screenshots**
<img width="1211" alt="Screen Shot 2020-10-03 at 8 09 08 PM" src="https://user-images.githubusercontent.com/7570704/95004699-5435db80-05b4-11eb-916f-1eb2c7762ee0.png">

**Logs**
```
DEBU[0559] received contextName: docker-desktop         
DEBU[0561] file successfully downloaded . . .           
DEBU[0561] getResource data: &unstructured.Unstructured{Object:map[string]interface {}{"metadata":map[string]interface {}{"name":"default"}}} 
DEBU[0561] getResource res: schema.GroupVersionResource{Group:"", Version:"v1", Resource:"namespaces"} 
INFO[0561] Retrieved Resource of type:  and name: default 
DEBU[0561] retrieved namespace: &unstructured.Unstructured{Object:map[string]interface {}{"apiVersion":"v1", "kind":"Namespace", "metadata":map[string]interface {}{"creationTimestamp":"2020-10-04T01:00:27Z", "managedFields":[]interface {}{map[string]interface {}{"apiVersion":"v1", "fieldsType":"FieldsV1", "fieldsV1":map[string]interface {}{"f:status":map[string]interface {}{"f:phase":map[string]interface {}{}}}, "manager":"kube-apiserver", "operation":"Update", "time":"2020-10-04T01:00:27Z"}}, "name":"default", "resourceVersion":"151", "selfLink":"/api/v1/namespaces/default", "uid":"335cef97-cf45-4545-a336-c323b927c3c6"}, "spec":map[string]interface {}{"finalizers":[]interface {}{"kubernetes"}}, "status":map[string]interface {}{"phase":"Active"}}} 
INFO[0561] Updated Resource of type: Namespace and name: default 
DEBU[0561] &{/v1, Resource=namespaces /v1, Kind=Namespace 0x2f01180} 
DEBU[0561] map[apiVersion:v1 kind:Namespace metadata:map[creationTimestamp:<nil> name:emojivoto] spec:map[] status:map[]] 
INFO[0561] Create the Namespace emojivoto succeed       
DEBU[0561] &{/v1, Resource=serviceaccounts /v1, Kind=ServiceAccount 0x2f01170} 
DEBU[0561] map[apiVersion:v1 kind:ServiceAccount metadata:map[creationTimestamp:<nil> name:emoji namespace:emojivoto]] 
INFO[0561] Create the ServiceAccount emoji in namespace emojivoto succeed 
DEBU[0561] &{/v1, Resource=serviceaccounts /v1, Kind=ServiceAccount 0x2f01170} 
DEBU[0561] map[apiVersion:v1 kind:ServiceAccount metadata:map[creationTimestamp:<nil> name:voting namespace:emojivoto]] 
INFO[0561] Create the ServiceAccount voting in namespace emojivoto succeed 
DEBU[0561] &{/v1, Resource=serviceaccounts /v1, Kind=ServiceAccount 0x2f01170} 
DEBU[0561] map[apiVersion:v1 kind:ServiceAccount metadata:map[creationTimestamp:<nil> name:web namespace:emojivoto]] 
INFO[0561] Create the ServiceAccount web in namespace emojivoto succeed 
DEBU[0561] &{/v1, Resource=services /v1, Kind=Service 0x2f01170} 
DEBU[0561] map[apiVersion:v1 kind:Service metadata:map[creationTimestamp:<nil> name:emoji-svc namespace:emojivoto] spec:map[ports:[map[name:grpc port:8080 targetPort:8080] map[name:prom port:8801 targetPort:8801]] selector:map[app:emoji-svc]] status:map[loadBalancer:map[]]] 
INFO[0561] Create the Service emoji-svc in namespace emojivoto succeed 
DEBU[0561] &{/v1, Resource=services /v1, Kind=Service 0x2f01170} 
DEBU[0561] map[apiVersion:v1 kind:Service metadata:map[creationTimestamp:<nil> name:voting-svc namespace:emojivoto] spec:map[ports:[map[name:grpc port:8080 targetPort:8080] map[name:prom port:8801 targetPort:8801]] selector:map[app:voting-svc]] status:map[loadBalancer:map[]]] 
INFO[0561] Create the Service voting-svc in namespace emojivoto succeed 
DEBU[0561] &{/v1, Resource=services /v1, Kind=Service 0x2f01170} 
DEBU[0561] map[apiVersion:v1 kind:Service metadata:map[creationTimestamp:<nil> name:web-svc namespace:emojivoto] spec:map[ports:[map[name:http port:80 targetPort:8080]] selector:map[app:web-svc] type:ClusterIP] status:map[loadBalancer:map[]]] 
INFO[0561] Create the Service web-svc in namespace emojivoto succeed 
DEBU[0562] &{apps/v1, Resource=deployments apps/v1, Kind=Deployment 0x2f01170} 
DEBU[0562] map[apiVersion:apps/v1 kind:Deployment metadata:map[creationTimestamp:<nil> labels:map[app.kubernetes.io/name:emoji app.kubernetes.io/part-of:emojivoto app.kubernetes.io/version:v10] name:emoji namespace:emojivoto] spec:map[replicas:1 selector:map[matchLabels:map[app:emoji-svc version:v10]] strategy:map[] template:map[metadata:map[creationTimestamp:<nil> labels:map[app:emoji-svc version:v10]] spec:map[containers:[map[env:[map[name:GRPC_PORT value:8080] map[name:PROM_PORT value:8801]] image:buoyantio/emojivoto-emoji-svc:v10 name:emoji-svc ports:[map[containerPort:8080 name:grpc] map[containerPort:8801 name:prom]] resources:map[requests:map[cpu:100m]]]] serviceAccountName:emoji]]] status:map[]] 
INFO[0562] Create the Deployment emoji in namespace emojivoto succeed 
DEBU[0562] &{apps/v1, Resource=deployments apps/v1, Kind=Deployment 0x2f01170} 
DEBU[0562] map[apiVersion:apps/v1 kind:Deployment metadata:map[creationTimestamp:<nil> labels:map[app.kubernetes.io/name:vote-bot app.kubernetes.io/part-of:emojivoto app.kubernetes.io/version:v10] name:vote-bot namespace:emojivoto] spec:map[replicas:1 selector:map[matchLabels:map[app:vote-bot version:v10]] strategy:map[] template:map[metadata:map[creationTimestamp:<nil> labels:map[app:vote-bot version:v10]] spec:map[containers:[map[command:[emojivoto-vote-bot] env:[map[name:WEB_HOST value:web-svc.emojivoto:80]] image:buoyantio/emojivoto-web:v10 name:vote-bot resources:map[requests:map[cpu:10m]]]]]]] status:map[]] 
INFO[0562] Create the Deployment vote-bot in namespace emojivoto succeed 
DEBU[0563] &{apps/v1, Resource=deployments apps/v1, Kind=Deployment 0x2f01170} 
DEBU[0563] map[apiVersion:apps/v1 kind:Deployment metadata:map[creationTimestamp:<nil> labels:map[app.kubernetes.io/name:voting app.kubernetes.io/part-of:emojivoto app.kubernetes.io/version:v10] name:voting namespace:emojivoto] spec:map[replicas:1 selector:map[matchLabels:map[app:voting-svc version:v10]] strategy:map[] template:map[metadata:map[creationTimestamp:<nil> labels:map[app:voting-svc version:v10]] spec:map[containers:[map[env:[map[name:GRPC_PORT value:8080] map[name:PROM_PORT value:8801]] image:buoyantio/emojivoto-voting-svc:v10 name:voting-svc ports:[map[containerPort:8080 name:grpc] map[containerPort:8801 name:prom]] resources:map[requests:map[cpu:100m]]]] serviceAccountName:voting]]] status:map[]] 
INFO[0563] Create the Deployment voting in namespace emojivoto succeed 
DEBU[0563] &{apps/v1, Resource=deployments apps/v1, Kind=Deployment 0x2f01170} 
DEBU[0563] map[apiVersion:apps/v1 kind:Deployment metadata:map[creationTimestamp:<nil> labels:map[app.kubernetes.io/name:web app.kubernetes.io/part-of:emojivoto app.kubernetes.io/version:v10] name:web namespace:emojivoto] spec:map[replicas:1 selector:map[matchLabels:map[app:web-svc version:v10]] strategy:map[] template:map[metadata:map[creationTimestamp:<nil> labels:map[app:web-svc version:v10]] spec:map[containers:[map[env:[map[name:WEB_PORT value:8080] map[name:EMOJISVC_HOST value:emoji-svc.emojivoto:8080] map[name:VOTINGSVC_HOST value:voting-svc.emojivoto:8080] map[name:INDEX_BUNDLE value:dist/index_bundle.js]] image:buoyantio/emojivoto-web:v10 name:web-svc ports:[map[containerPort:8080 name:http]] resources:map[requests:map[cpu:100m]]]] serviceAccountName:web]]] status:map[]] 
INFO[0563] Create the Deployment web in namespace emojivoto succeed 
DEBU[0563] getResource data: &unstructured.Unstructured{Object:map[string]interface {}{"metadata":map[string]interface {}{"name":"web-svc", "namespace":"default"}}} 
DEBU[0563] getResource res: schema.GroupVersionResource{Group:"", Version:"v1", Resource:"services"} 
WARN[0563] unable to retrieve the resource with a matching name, attempting operation without namespace: services "web-svc" not found 
ERRO[0563] unable to retrieve the resource with a matching name, while attempting to apply the config: the server could not find the requested resource 
ERRO[0563] unable to get service details: unable to retrieve the resource with a matching name, while attempting to apply the config: the server could not find the requested resource 
DEBU[0563] sending event: &meshes.EventsResponse{EventType:1, Summary:"Emojivoto App is deployed but unable to retrieve the port info for the service at the moment", Details:"unable to get service details: unable to retrieve the resource with a matching name, while attempting to apply the config: the server could not find the requested resource", OperationId:"e65a9529-35f4-48b9-bf4e-231dcd0acec9", XXX_NoUnkeyedLiteral:struct {}{}, XXX_unrecognized:[]uint8(nil), XXX_sizecache:0} 
```

**Comment by utkarsh-pro** [comment]
@leecalcote after searching around for a while I found that here
https://github.com/layer5io/meshery-linkerd/blob/f2aff92347096940739df3a546b684883e14fe1e/linkerd/linkerd.go#L488-L490

we are overriding the `namespace` that was received and end up creating the resource in the namespace from the YAML file. But when we search for the resource (to look for the port), we search in the namespace given by the user. This is why the deployment is succeeding but port lookup fails.

I removed L488-L490 as well as changed 
https://github.com/layer5io/meshery-linkerd/blob/f2aff92347096940739df3a546b684883e14fe1e/linkerd/linkerd.go#L546
to
```diff
- if deleteOpts && dataNamespace.GetName() != "default" {
+ if deleteOpts && dataNamespace.GetNamespace() != "default" {
```

so that it doesn't attempt delete the default namespace and it seemed to fix the issue. Not creating a PR as I am not sure if these changes will break anything :sweat_smile: .
